### DISCUSSION TITLE
How do i generate (way) less variations?

### DISCUSSION THREAD

**Discussion by jackmcdade** [spatie/statamic-responsive-images#254] [Q&A] [answered]
It's way too memory intensive to generate 15-30 variations per image, not to mention a huge amount of storage space. How can I generate let's say 3-5 variations MAX? Is this possible? I've been trying to figure this out for hours and all I'm running into is server crashes and am honestly at my wits end trying to make this work for a new Starter Kit we're launching.

**Comment by ncla** [comment]
Hi Jack, it's been a while I have developed for Statamic and this package, but quite some time ago I implemented a change that allows for the dimension calculations to be overridden. https://github.com/spatie/statamic-responsive-images/pull/193

This should allow you to cut down on the variants produced.

You would have to bind your own `DimensionCalculator` in a service provider or else where similar. https://github.com/spatie/statamic-responsive-images/blob/main/src/ResponsiveDimensionCalculator.php

**Reply by jackmcdade** [reply]
Ah gotcha — no documentation on this then, just a code dive?

**Reply by ncla** [reply]
Yup, I was not that confident to document it at the time. But it seems Rias was kind and fast enough to solve your base need. :) So no need to code dive, unless you want more control still.

**Comment by riasvdv** [comment]
One thing you can do is set the [max width](https://github.com/spatie/statamic-responsive-images/blob/main/config/responsive-images.php#L62) config, which will give an upper limit to the image sizes being generated.

We also don't advise using glide image caching in combination with this addon, as it generates all the variations upfront, turning it off will only generate the variations that are necessary (usually optimized for the most common screen sizes)

A final thing you can do if you really only want a few sizes, is to implement your own `DimensionCalculator` and bind it instead of the default one https://github.com/spatie/statamic-responsive-images/blob/main/src/ServiceProvider.php#L108

**Reply by jackmcdade** [reply]
I've found that without caching the images and generating all the variations up front, it will crash servers. I've spent 8 hours trying to stand up a simple photo gallery and it brought my server down over a dozen times, even with Horizon and 4GB ram. Something ain't right. 🤔

**Reply by riasvdv** [reply, marked as answer]
@jackmcdade I've just added the threshold for the default DimensionCalculator to the config file: https://github.com/spatie/statamic-responsive-images/blob/main/config/responsive-images.php#L76

Setting this lower will result in less variants being generated. You'll need to update to the latest version for this

**Reply by jackmcdade** [reply]
Now we're talkin! Thank you!

**Reply by johncarter-** [reply]
This is why I'm always crying about images! Maybe this solves my issues too, thanks Rias, and Jack for figuring out what was happening.
